service/src/main/kotlin/app/cash/backfila/BackfillCreator.kt

```diff
@@ -0,0 +1,183 @@
+package app.cash.backfila
+
+import app.cash.backfila.client.ConnectorProvider
+import app.cash.backfila.protos.clientservice.KeyRange
+import app.cash.backfila.protos.clientservice.PrepareBackfillRequest
+import app.cash.backfila.protos.clientservice.PrepareBackfillResponse
+import app.cash.backfila.protos.service.CreateBackfillRequest
+import app.cash.backfila.service.persistence.BackfilaDb
+import app.cash.backfila.service.persistence.BackfillState
+import app.cash.backfila.service.persistence.DbBackfillRun
+import app.cash.backfila.service.persistence.DbRegisteredBackfill
+import app.cash.backfila.service.persistence.DbRunPartition
+import app.cash.backfila.service.persistence.DbService
+import app.cash.backfila.service.persistence.RegisteredBackfillQuery
+import app.cash.backfila.service.persistence.ServiceQuery
+import misk.exceptions.BadRequestException
+import misk.hibernate.Id
+import misk.hibernate.Query
+import misk.hibernate.Transacter
+import misk.hibernate.newQuery
+import misk.logging.getLogger
+import javax.inject.Inject
+
+class BackfillCreator @Inject constructor(
+    @BackfilaDb private val transacter: Transacter,
+    private val queryFactory: Query.Factory,
+    private val connectorProvider: ConnectorProvider
+) {
+  fun create(
+      author: String,
+      service: String,
+      request: CreateBackfillRequest
+  ): Id<DbBackfillRun> {
+    logger.info { "Create backfill for `$service` by `$author`" }
```

Comment:
Is author required? Sometimes it might not make sense?

Reply:
we always have one, for now at least

Reply:
I guess you can always fake with a service user or something.